src/nunit.analyzers/NonTestMethodAccessibilityLevel/NonTestMethodAccessibilityLevelAnalyzer.cs

```diff
@@ -79,5 +86,20 @@ private static bool IsPublicOrInternalMethod(IMethodSymbol method)
                     return false;
             }
         }
+
+        private static bool IsDisposeMethod(IMethodSymbol method)
+        {
+            if (method.IsOverride)
+            {
+                return IsDisposeMethod(method.OverriddenMethod);
+            }
+
+            INamedTypeSymbol? type = method.ContainingType;
+            var disposeInterfaceMethods = type.AllInterfaces
+                    .Where(x => x.SpecialType == SpecialType.System_IDisposable)
+                    .SelectMany(i => i.GetMembers().OfType<IMethodSymbol>());
```

Comment:
Reading the code it seems a bit odd that `type` is typed as nullable and we then in the following line dereference it. Should this be `type?.AllInterfaces.Where(....).SelectMany(...) ?? Array.Empty<IMethodSymbol>()`?

Reply:
You are right. The problem is that we have nullable enabled but the Roslyn nuget package we reference has no nullability attributes, so I don't get the warnings in the compiler.  I will see if we can upgrade to a later package (in a separate PR). Are we constrained to netstandard1.6 in order to support VS2017?

Although I think in this case the `ContainingType` will never be null as the symbol comes from `type.GetMembers()`.
To be safe for future code changes, I will add a condition to bail out if type is unexpectedly null.

Reply:
Yes. The reason for netstandard1.6 is to still support VS 2017, but I'm considering if we should drop it. However, we still see some reports in the NUnit repo from people still on 2017 (and even 2015). Alternatives could be multitargeting or separate projects/repos, but both seem somewhat overkill. 

I'll give it some more thought.